packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart

```diff
@@ -15,9 +23,270 @@ class DiffPane extends StatefulWidget {
+        );
+      },
+    );
+  }
+}
+
+class _SnapshotListTitle extends StatelessWidget {
+  const _SnapshotListTitle({
+    Key? key,
+    required this.item,
+    required this.selected,
+  }) : super(key: key);
+  final DiffListItem item;
+  final bool selected;
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: item,
+      builder: (context, child) => Row(
+        children: [
+          _SelectionBox(selected: selected),
+          const SizedBox(width: denseRowSpacing),
+          if (item.isProcessing) const _ProgressIndicator(),
+          if (item.isProcessing) const SizedBox(width: denseRowSpacing),
+          Expanded(
+            child: Text(item.name, overflow: TextOverflow.ellipsis),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ProgressIndicator extends StatelessWidget {
+  const _ProgressIndicator({Key? key}) : super(key: key);
+  static const _progressIndicatorSize = 16.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: _progressIndicatorSize,
+      height: _progressIndicatorSize,
+      child: CircularProgressIndicator(
+        color: Theme.of(context).textTheme.bodyText1?.color,
+      ),
+    );
+  }
+}
+
+/// Blue or transparent square, to mark selected item.
+class _SelectionBox extends StatelessWidget {
+  const _SelectionBox({Key? key, required this.selected}) : super(key: key);
+  static const _boxSize = Size(3.0, 40.0);
+  final bool selected;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: _boxSize.width,
+      height: _boxSize.height,
+      child: selected
+          ? const DecoratedBox(
+              decoration: BoxDecoration(
+                color: defaultSelectionColor,
+              ),
+            )
+          : null,
+    );
+  }
+}
+
+class _SnapshotListContent extends StatelessWidget {
+  const _SnapshotListContent({Key? key, required this.item}) : super(key: key);
+  final DiffListItem item;
+
+  @override
+  Widget build(BuildContext context) {
+    final itemLocal = item;
+    if (itemLocal is InformationListItem)
+      return const Text('Introduction to snapshot diffing will be here.');
+    if (itemLocal is SnapshotListItem)
+      return Text('Content of ${itemLocal.name} will be here.');
+    throw 'Unexpected type of the item: ${itemLocal.runtimeType}';
+  }
+}
+
+class _ListControlPane extends StatelessWidget {
+  const _ListControlPane({Key? key, required this.controller})
+      : super(key: key);
+
+  final _DiffPaneController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<bool>(
+      valueListenable: controller.isProcessing,
+      builder: (_, isProcessing, __) {
+        final showTakeSnapshot = !isProcessing;
+        final showClearAll = !isProcessing & controller.hasSnapshots;
+        return Row(
+          children: [
+            ToolbarAction(
```

Comment:
since we have room for it, maybe we should add the text 'Take Heap Snapshot' or 'Take Snapshot' to make it more clear?

Reply:
I feel it will make the UI more noisy, i.e. it will be harder to focus on important things.
The informational list item on the top will explain what to click to record snapshot (the icon will be included). Also, there is the tooltip.
User will need to learn it just once.